File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/go_solutions/154-find-minimum-in-rotated-sorted-array-ii.go
func findMin(nums []int) int {
    low := 0
    high := len(nums) - 1

    for low < high {
        mid := low + (high-low)/2
        if nums[mid] < nums[high] {
            high = mid
        } else if nums[mid] > nums[high] {
            low = mid + 1
        } else { // nums[mid] == nums[high]
            high--
        }
    }

    return nums[low]
}

File: leetcode/AIcoding/3.DeepSeek-v3/erlang_solutions/154-find-minimum-in-rotated-sorted-array-ii.erl
-spec find_min(Nums :: [integer()]) -> integer().
find_min(Nums) ->
    find_min(Nums, 0, length(Nums) - 1).

find_min([H|_], Low, High) when Low >= High ->
    H;
find_min(Nums, Low, High) ->
    Mid = (Low + High) div 2,
    case lists:nth(Mid + 1, Nums) > lists:nth(High + 1, Nums) of
        true -> find_min(Nums, Mid + 1, High);
        false ->
            case lists:nth(Mid + 1, Nums) < lists:nth(High + 1, Nums) of
                true -> find_min(Nums, Low, Mid);
                false -> find_min(Nums, Low, High - 1)
            end
    end.